### PR TITLE
Auto-detect GITHUB_TOKEN env var

### PR DESCRIPTION
**Pull request by aaronsteers**
Resolves #25

### CODE REVIEW

**Comment by aaronsteers** [comment]
Not yet working:

`time=2021-09-23 20:34:09 name=tap-github level=INFO message=No auth token detected. For higher rate limits, please specify `auth_token` in config.
`

File: .github/workflows/test_tap.yml [hunk 1]
```diff
@@ -32,4 +32,4 @@ jobs:
         poetry run mypy . --ignore-missing-imports
     - name: Test with pytest
       run: |
-        poetry run pytest
+        poetry run pytest --capture=no
```
Comment (ericboucher) [inline .github/workflows/test_tap.yml:35]: I think capture can be useful to debug on the fly in the CI. Any strong reason for silencing?

**Comment by ericboucher** [comment]
@aaronsteers this is working locally. For it to work in GitHub action, you will simply need to map them explicitly. See https://medium.datadriveninvestor.com/accessing-github-secrets-in-python-d3e758d8089b

So just adding this to your action definition should work:

```
env:
    GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
```

**Comment by aaronsteers** [comment]
> @aaronsteers this is working locally for me. For it to work in GitHub action, you will simply need to map them explicitly. See https://medium.datadriveninvestor.com/accessing-github-secrets-in-python-d3e758d8089b
> 
> So just adding this to your action definition should work:
> 
> ```
> env:
>     GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
> ```

@ericboucher  That worked!!

> time=2021-09-24 15:49:15 name=tap-github level=INFO message=Found 'GITHUB_TOKEN' environment variable for authentication.

File: .github/workflows/test_tap.yml (hunk 1 above)
Comment (aaronsteers) [reply .github/workflows/test_tap.yml:37]: @ericboucher - This one is confusingly named. "Not capturing" means "not suppressing". Added this setting allowed me to observe the log output needed to confirm the auth.

**Comment by aaronsteers** [comment]
@ericboucher - If you don't mind being my second pair of eyes, here, would be much appreciated. (PR will auto-merge if approved.)

File: .github/workflows/test_tap.yml (hunk 1 above)
Comment (ericboucher) [reply .github/workflows/test_tap.yml:37]: 🤦

File: tap_github/client.py [hunk 2]
```diff
@@ -28,8 +29,20 @@ def http_headers(self) -> dict:
         headers = {"Accept": "application/vnd.github.v3+json"}
         if "user_agent" in self.config:
             headers["User-Agent"] = cast(str, self.config.get("user_agent"))
+
```
Comment (ericboucher) [inline tap_github/client.py:32]: No need for this newline I think

File: tap_github/client.py [hunk 3]
```diff
@@ -28,8 +29,20 @@ def http_headers(self) -> dict:
...
         if "auth_token" in self.config:
             headers["Authorization"] = f"token {self.config['auth_token']}"
+        elif "GITHUB_TOKEN" in environ:
+            self.logger.info(
+                "Found 'GITHUB_TOKEN' environment variable for authentication."
```
Comment (ericboucher) [inline tap_github/client.py:37]: Being picky but I would say `Using 'GITHUB_TOKEN'`

**Review by ericboucher — APPROVED** [review]
Looks great! Two minor comments but I can address them in my open PR if you agree.

**Comment by ericboucher** [comment]
@aaronsteers looks like my approval doesn't have enough power 😉

**Pull request merged**